crates/test-collector/src/lib.rs

```diff
@@ -64,17 +63,13 @@ pub fn setup_project_without_cairo_project_toml(
     path: &Path,
     crate_name: &str,
 ) -> Result<Vec<CrateId>, ProjectError> {
-    if path.is_dir() {
-        match build_project_config(path, crate_name) {
-            Ok(config) => {
-                let main_crate_ids = get_main_crate_ids_from_project(db, &config);
-                update_crate_roots_from_project_config(db, config);
-                Ok(main_crate_ids)
-            }
-            _ => Err(ProjectError::LoadProjectError),
+    match build_project_config(path, crate_name) {
+        Ok(config) => {
+            let main_crate_ids = get_main_crate_ids_from_project(db, &config);
+            update_crate_roots_from_project_config(db, config);
+            Ok(main_crate_ids)
         }
-    } else {
-        Ok(vec![setup_single_file_project(db, path)?])
+        _ => Err(ProjectError::LoadProjectError),
```

Comment:
I think we should panic here instead, if we pass a non dir path it is a bug on our side

Reply:
It's error handling from the build_project_config function, but I can add an assert if path is a directory.